src/test/java/teammates/sqllogic/core/UsersLogicTest.java

```diff
@@ -0,0 +1,151 @@
+import teammates.storage.sqlapi.UsersDb;
+import teammates.storage.sqlentity.Account;
+import teammates.storage.sqlentity.Course;
+import teammates.storage.sqlentity.Instructor;
+import teammates.storage.sqlentity.Student;
+import teammates.test.BaseTestCase;
+
+/**
+ * SUT: {@link UsersLogic}.
+ */
+public class UsersLogicTest extends BaseTestCase {
+
+    private UsersLogic usersLogic = UsersLogic.inst();
+
+    private AccountsLogic accountsLogic;
+
+    private UsersDb usersDb;
+
+    private Course course;
+
+    @BeforeMethod
+    public void setUpMethod() {
+        usersDb = mock(UsersDb.class);
+        accountsLogic = mock(AccountsLogic.class);
+        usersLogic.initLogicDependencies(usersDb, accountsLogic);
+
+        course = new Course("course-id", "course-name", Const.DEFAULT_TIME_ZONE, "institute");
+    }
+
+    @Test
+    public void testResetInstructorGoogleId_entityExistsWithNoUserAssociatedWithGoogleId_success()
+            throws EntityDoesNotExistException {
+        Account account = generateTypicalAccount();
+        Instructor instructor = getTypicalInstructor();
+        String courseId = instructor.getCourseId();
+        String email = instructor.getEmail();
+
+        instructor.setAccount(account);
+
+        String googleId = instructor.getAccount().getGoogleId();
+
+        when(usersLogic.getInstructorForEmail(courseId, email)).thenReturn(instructor);
+        when(usersDb.getAllUsersByGoogleId(googleId)).thenReturn(Collections.emptyList());
+        when(accountsLogic.getAccountForGoogleId(googleId)).thenReturn(account);
+
+        usersLogic.resetInstructorGoogleId(email, courseId, googleId);
+
+        assertEquals(null, instructor.getAccount());
+        verify(accountsLogic, times(1)).deleteAccountCascade(googleId);
+    }
+
+    @Test
+    public void testResetInstructorGoogleId_entityDoesNotExists_throwsEntityDoesNotExistException()
+            throws EntityDoesNotExistException {
+        Account account = generateTypicalAccount();
+        Instructor instructor = getTypicalInstructor();
+        String courseId = instructor.getCourseId();
+        String email = instructor.getEmail();
+
+        instructor.setAccount(account);
+
+        String googleId = instructor.getAccount().getGoogleId();
+
+        when(usersLogic.getInstructorForEmail(courseId, email)).thenReturn(null);
+
+        EntityDoesNotExistException exception = assertThrows(EntityDoesNotExistException.class,
+                () -> usersLogic.resetInstructorGoogleId(email, courseId, googleId));
+
+        assertEquals(ERROR_UPDATE_NON_EXISTENT
+                + "Instructor [courseId=" + courseId + ", email=" + email + "]", exception.getMessage());
+    }
+
+    @Test
+    public void testResetStudentGoogleId_entityExistsWithNoUserAssociatedWithGoogleId_success()
+            throws EntityDoesNotExistException {
+        Account account = generateTypicalAccount();
+        Student student = getTypicalStudent();
```

Comment:
These should be a method field since it is used everywhere. Unit tests should ideally have as little setup as possible.